xdcr/xdcr_test.go

```diff
@@ -50,74 +52,299 @@ func TestMobileXDCRNoSyncDataCopied(t *testing.T) {
+				_, err := toDs.WriteCas(docID, 0, 0, []byte(`{"datastore":"toDs"}`), 0)
+				require.NoError(t, err)
+				cas, err := fromDs.WriteCas(docID, 0, 0, []byte(`{"datastore":"fromDs"}`), 0)
+				require.NoError(t, err)
+				return cas
+			},
+		},
+		{
+			name:  "dest doc newer, expect keep old version",
+			docID: "doc3",
+			body:  `{"datastore":"fromDs"}`,
+			preXDCRFunc: func(t *testing.T, docID string) uint64 {
+				_, err := toDs.WriteCas(docID, 0, 0, []byte(`{"datastore":"toDs"}`), 0)
+				require.NoError(t, err)
+				cas, err := fromDs.WriteCas(docID, 0, 0, []byte(`{"datastore":"fromDs"}`), 0)
+				require.NoError(t, err)
+				return cas
+			},
+		},
 	}
+	// tests write a document
+	// start xdcr
+	// verify result
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			fromCAS := testCase.preXDCRFunc(t, testCase.docID)
+
+			xdcr := startXDCR(t, fromBucket, toBucket, XDCROptions{Mobile: MobileOn})
+			defer func() {
+				assert.NoError(t, xdcr.Stop(ctx))
+			}()
+			requireWaitForXDCRDocsWritten(t, xdcr, 1)
+
+			body, xattrs, destCas, err := toDs.GetWithXattrs(ctx, testCase.docID, []string{base.VvXattrName, base.MouXattrName})
+			require.NoError(t, err, "Could not get doc %s", testCase.docID)
+			require.Equal(t, fromCAS, destCas)
+			require.JSONEq(t, testCase.body, string(body))
+			require.NotContains(t, xattrs, base.MouXattrName)
+			require.Contains(t, xattrs, base.VvXattrName)
+			casString := string(base.Uint64CASToLittleEndianHex(fromCAS))
+			var vv *db.HybridLogicalVector
+			require.NoError(t, base.JSONUnmarshal(xattrs[base.VvXattrName], &vv))
+			require.Equal(t, &db.HybridLogicalVector{
+				CurrentVersionCAS: casString,
+				SourceID:          fromBucketSourceID,
+				Version:           casString,
+			}, vv)
+
+		})
+	}
+}
+
+func TestVVWriteTwice(t *testing.T) {
+	fromBucket, fromDs, toBucket, toDs := getTwoBucketDataStores(t)
+	ctx := base.TestCtx(t)
+	fromBucketSourceID, err := getSourceID(ctx, fromBucket)
+	require.NoError(t, err)
+
+	docID := "doc1"
+	ver1Body := `{"ver":1}`
+	fromCAS, err := fromDs.WriteCas(docID, 0, 0, []byte(ver1Body), 0)
+	require.NoError(t, err)
+	xdcr := startXDCR(t, fromBucket, toBucket, XDCROptions{Mobile: MobileOn})
+	defer func() {
+		assert.NoError(t, xdcr.Stop(ctx))
+	}()
+	requireWaitForXDCRDocsWritten(t, xdcr, 1)
+
+	body, xattrs, destCas, err := toDs.GetWithXattrs(ctx, docID, []string{base.VvXattrName, base.MouXattrName})
+	require.NoError(t, err)
+	require.Equal(t, fromCAS, destCas)
+	require.JSONEq(t, ver1Body, string(body))
+	require.Contains(t, xattrs, base.VvXattrName)
+	casString := string(base.Uint64CASToLittleEndianHex(fromCAS))
+	var vv *db.HybridLogicalVector
+	require.NoError(t, base.JSONUnmarshal(xattrs[base.VvXattrName], &vv))
+	require.Equal(t, &db.HybridLogicalVector{
+		CurrentVersionCAS: casString,
+		SourceID:          fromBucketSourceID,
+		Version:           casString,
+	}, vv)
```

Comment:
This seems like commonly reused code.  Maybe add a toDs.VerifyCV(t, cvcas, sourceID, version) function?  (which could call a toDS.VerifyHLV(t, HLV) for future tests that care about pv?